Content.Shared/GameObjects/Components/Weapons/Ranged/Barrels/SharedBatteryBarrelComponent.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Robust.Shared.GameObjects;
+using Robust.Shared.Serialization;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Content.Shared.GameObjects.Components.Weapons.Ranged.Barrels
+{
+    [Serializable, NetSerializable]
+    public class BatteryBarrelComponentState : ComponentState
+    {
+        public FireRateSelector FireRateSelector { get; }
+        public (int count, int max)? Magazine { get; }
+        public string SoundGunshot { get; }
```

Comment:
This seems unused on the client. What does the client need it for?

Reply:
Nothing :fingerguns:
I just copied the shit over. The SoundGunshot can be removed. I'm not too sure for the FireRateSelector (we don't have any weapons that can switch firerate **yet**)

Reply:
I guess you can leave FireRateSelector for now.